sim/dex.js

```diff
@@ -964,9 +964,9 @@ class ModdedDex {
 	}
 
 	/**
-	 * TODO: TypeScript generics
-	 * @param {Array} arr
-	 * @return {Array}
+	 * @param {T[]} arr
+	 * @return {T[]}
+	 * @template T
```

Comment:
Yeah, TypeScript generics didn't work a yearish ago; I guess they finally fixed this since I've been seeing generics recently.